analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,179 @@
+
+        private static readonly int[] InvalidLogLevel =
+        {
+            0, // Trace
+            1, // Debug
+            6, // None
+        };
+
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var catchClause = (CatchClauseSyntax)c.Node;
+                if (c.AzureFunctionMethod() is { } entryPoint
+                    && HasLoggerInScope(entryPoint, c.CancellationToken))
+                {
+                    var walker = new LoggerCallWalker(c.SemanticModel, c.CancellationToken);
+                    walker.SafeVisit(catchClause.Block);
+                    // Exception handling in the filter clause preserves log scopes and is therefore recommended
+                    // See https://blog.stephencleary.com/2020/06/a-new-pattern-for-exception-logging.html
+                    walker.SafeVisit(catchClause.Filter?.FilterExpression);
+                    if (!walker.HasValidLoggerCall)
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, catchClause.CatchKeyword.GetLocation(), walker.InvalidLoggerInvocationLocations));
+                    }
+                }
+            },
+            SyntaxKind.CatchClause);
+
+        private static bool HasLoggerInScope(IMethodSymbol entryPoint, CancellationToken cancellationToken) =>
+            entryPoint.Parameters.Any(x => x.Type.DerivesOrImplements(KnownType.Microsoft_Extensions_Logging_ILogger))
+                // Instance method entry points might have access to an ILogger via injected fields/properties
+                // https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection
+                || (entryPoint is { IsStatic: false, ContainingType: { } container }
+                    && HasLoggerMember(container, cancellationToken) is not null);
+
+        internal static ISymbol HasLoggerMember(ITypeSymbol typeSymbol, CancellationToken cancellationToken)
+        {
+            var isOriginalType = true;
+            foreach (var type in typeSymbol.GetSelfAndBaseTypes())
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                if (type.GetMembers().FirstOrDefault(x => x.Kind is SymbolKind.Field or SymbolKind.Property
+                                                          && AccessibilityValid(x.GetEffectiveAccessibility(), isOriginalType)
+                                                          && x.GetSymbolType()?.DerivesOrImplements(KnownType.Microsoft_Extensions_Logging_ILogger) is true) is { } result)
+                {
+                    return result;
```

Comment:
We no longer need that object. So we can change the `FirstOrDefault` to `Any` now
```suggestion
                    return true;
```